04-onion-routing.md

```diff
@@ -367,6 +368,96 @@ otherwise meets the amount criterion (eg. some other failure, or
 invoice timeout), however if it were to fulfill only some of them,
 intermediary nodes could simply claim the remaining ones.
 
+### Onion Messages
+
+Onion messages have an onion with an alternate `hop_payload`
+format: a `bigsize` followed by a `onionmsg_payload`.  Note that there
+is no legacy format, thus a `bigsize` of 0 means no payload.
+
+1. tlvs: `onionmsg_payload`
+2. types:
+    1. type: 4 (`next_node_id`)
```

Comment:
To avoid a network-wide upgrade for onion message passing, could an intermediary step be to implement offers with only direct messages?

Reply:
Yes, in fact (unless suppressed), c-lightning will connect directly if there's no path found which supports the features.